pkg/util/cluster/aad.go

```diff
@@ -66,11 +53,12 @@ func (c *Cluster) createServicePrincipal(ctx context.Context, appID string) (str
 	// wait.PollImmediateUntil. Doing this will not propagate the latest error
 	// to the user in case when wait exceeds the timeout
 	_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
-		sp, err = c.serviceprincipals.Create(ctx, azgraphrbac.ServicePrincipalCreateParameters{
-			AppID: &appID,
-		})
-		if detailedErr, ok := err.(autorest.DetailedError); ok &&
-			detailedErr.StatusCode == http.StatusForbidden {
+		requestBody := msgraph_models.NewServicePrincipal()
+		requestBody.SetAppId(&appID)
+		result, err = c.spGraphClient.ServicePrincipals().Post(ctx, requestBody, nil)
+
+		if oDataError, ok := err.(msgraph_errors.ODataErrorable); ok &&
+			*oDataError.GetError().GetCode() == "accessDenied" {
```

Comment:
Will msft graph ever give us an App doesn't exist error until the application is replicated?

Reply:
I don't know yet.  It's hard to predict what the failure behavior will be until we see it in the wild.